provisioner/terraform/modules_internal_test.go

```diff
@@ -0,0 +1,18 @@
+package terraform
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetModulesArchive(t *testing.T) {
+	t.Parallel()
+
+	archive, err := getModulesArchive("testdata/modules-source-caching")
+	require.NoError(t, err)
+	hash := sha256.Sum256(archive)
+	require.Equal(t, "0ac7b7b3ff92d1e4bfd7ea1bef64fd7d7ac40434409fac158e383dcdd5ebeb73", hex.EncodeToString(hash[:]))
+}
```

Comment:
This test is failing.

I wonder if it would be better to compare the output to hash of an actual `output.tar` file in `testdata`. That way we can manually inspect the tar for what it looks like.

A hash compare is fine for now, but a bit tricky to maintain if something changes. Someone will just update the test to a new arbitrary hash.

-----

That or we open the tar in the test and assert some files exist.

Reply:
done!